src/domain/hanke/edit/utils.ts

```diff
@@ -108,3 +110,24 @@ export function calculateTotalSurfaceArea(areas?: HankeAlueFormState[]) {
 export function canHankeBeCancelled(applications: Application[]): boolean {
   return applications.every((application) => isApplicationPending(application.alluStatus));
 }
+
+/**
+ * Get default name for hanke area
+ */
+export function getAreaDefaultName(areas?: HankeAlueFormState[]) {
+  if (areas === undefined) {
+    return undefined;
+  }
+
+  function getAreaNumber(area?: HankeAlueFormState): number {
+    const areaNumber = area?.nimi?.match(/\d+$/);
+    return areaNumber ? Number(areaNumber[0]) : 0;
+  }
+
+  const defaultNameRegExp = /^Hankealue \d+$/;
+  const areasWithDefaultName = areas.filter((area) => defaultNameRegExp.test(area.nimi || ''));
+  const areaWithMaxName = maxBy(areasWithDefaultName, (area) => getAreaNumber(area));
+  const maxAreaNumber: number = getAreaNumber(areaWithMaxName);
```

Comment:
It feels clearer to me to map the areas with a default name to their number and then to choose the max number.
```suggestion
  const maxAreaNumber = areas
    .filter((area) => defaultNameRegExp.test(area.nimi || ''))
    .map(getAreaNumber)
    .reduce((a, b) => Math.max(a, b), 0);
```

Reply:
You're right, that is clearer.